pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -425,46 +426,58 @@ public void sendAsync(Message<?> message, SendCallback callback) {
             return;
         }
 
-        if (!populateMessageSchema(msg, callback)) {
```

Comment:
is it necessary to change ProducerImpl ?

Reply:
same as above.